include/swift/AST/DiagnosticEngine.h

```diff
@@ -1053,6 +1057,20 @@ namespace swift {
     }
   };
 
+  /// Remember details about the state of a diagnostic engine. Use this when a
+  /// function that may be called while its caller is emitting diagnostics may
+  /// itself emit diagnostics--especially ignored diagnostics--and you want to
+  /// be sure the caller's diagnostics aren't affected.
```

Comment:
The second sentence here seems a bit difficult to read/understand (maybe because it's long?). How about something like (for example):

```
    /// Remember details about the state of a diagnostic engine.
    ///
    /// Typically, there is at most one in-flight diagnostic at any given time. This type
    /// helps work around that restriction; use it when all of the following are true:
    /// - Some code needs to emit a diagnostic (especially an ignored diagnostic).
    /// - The caller may be emitting diagnostics.
    /// - You want to be sure the caller's diagnostics aren't affected.
```

Reply:
This isn’t actually about in-flight diagnostics; it’s about notes being correctly associated with the previous diagnostic. I’ll change the comment to explain it better.